src/ai/susi/server/api/cms/FeedbackSkillService.java

```diff
@@ -0,0 +1,313 @@
+                            }
+                        }
+                        
+                        Timestamp timestamp = new Timestamp(System.currentTimeMillis());
+                        feedbackObject.put("email", email);
+                        feedbackObject.put("feedback", skill_feedback);
+                        feedbackObject.put("timestamp", timestamp.toString());
+                        if (!feedbackUpdated) {
+                        	skillName.put(feedbackObject);
+                        }
+
+                        languageName.put(skill_name, skillName);
+                        groupName.put(language_name, languageName);
+                        modelName.put(group_name, groupName);
+                        feedbackSkill.put(model_name, modelName, true);
+                        result.put("accepted", true);
+                        result.put("message", "Skill feedback updated");
+                        result.put("feedback", skill_feedback);
+                        return new ServiceResponse(result);
+                    }
+                }
+            }
+
+            addToSkillRatingJSON(call);
+
+            languageName.put(skill_name, createFeedbackArray(email, skill_feedback));
+            groupName.put(language_name, languageName);
+            modelName.put(group_name, groupName);
+            feedbackSkill.put(model_name, modelName, true);
+            result.put("accepted", true);
+            result.put("message", "Skill feedback updated");
+            result.put("feedback", skill_feedback);
+            return new ServiceResponse(result);
+
+        } else {
+            throw new APIException(422, "Access token not given.");
+        }
+
+
+    }
+
+    /* Utility function*/
+    public JSONArray createFeedbackArray(String email, String skill_feedback) {
+        JSONArray skillName = new JSONArray();
+        JSONObject feedbackObject = new JSONObject();
+        Timestamp timestamp = new Timestamp(System.currentTimeMillis());
+        feedbackObject.put("email", email);
+        feedbackObject.put("feedback", skill_feedback);
+        feedbackObject.put("timestamp", timestamp.toString());
+        skillName.put(feedbackObject);
+
+        return skillName;
+    }
+
+
+    public JSONObject createRatingObject() {
+        JSONObject skillName = new JSONObject();
+        skillName.put("positive", "0");
+        skillName.put("negative", "0");
+        skillName.put("feedback_count", "1");
+
+        JSONObject skillStars = new JSONObject();
+
+        skillStars.put("one_star", "0");
+        skillStars.put("two_star", "0");
+        skillStars.put("three_star", "0");
+        skillStars.put("four_star", "0");
+        skillStars.put("five_star", "0");
+        skillStars.put("avg_star", "0");
+        skillStars.put("total_star", "0");
+        
+        skillName.put("stars", skillStars);
+
+        return skillName;
+    }
+    
+    public void updateSkillRatingsJSON(Query call) {
```

Comment:
Could you please explain what's happening in this method and the method called `addToSkillRating`. It looks like only the feedback count increases and doesn't update the star rating details.

Reply:
Have changed the code entirely and refactored it. It now works for all cases. Earlier, it missed few cases like not working for all Groups.